Controller/NavbarController.php

```diff
@@ -14,22 +14,39 @@
 use KevinPapst\AdminLTEBundle\Event\ShowUserEvent;
 use KevinPapst\AdminLTEBundle\Event\TaskListEvent;
 use KevinPapst\AdminLTEBundle\Event\ThemeEvents;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 use Symfony\Component\HttpFoundation\Response;
 
 class NavbarController extends EmitterController
 {
-    public const MAX_NOTIFICATIONS = 5;
-    public const MAX_MESSAGES = 5;
-    public const MAX_TASKS = 5;
+    protected $max_notifications;
+    protected $max_messages;
+    protected $max_tasks;
 
-    public function notificationsAction($max = self::MAX_NOTIFICATIONS)
+    public function __construct(EventDispatcherInterface $dispatcher)
```

Comment:
```php
    public function __construct(EventDispatcherInterface $dispatcher, ContextHelper $helper)
    {
        parent::__construct($dispatcher);
        $this->max_notifications = $helper->getOption('max_navbar_notifications');
        ...
    }
```
if the container complains about missing dependency, you might have to add this to the `services.yml`:
```yaml
    KevinPapst\AdminLTEBundle\Helper\ContextHelper:
        alias: admin_lte_theme.context_helper
```

Reply:
> You have to adjust the KevinPapst\AdminLTEBundle\DependencyInjection\Configuration class for the new options, but I guess you know that...

Don't be so sure  :-P

Why do we need the context helper ? Is it related to the configuration ?

Reply:
The Bundle extension [sets the container parameter](https://github.com/kevinpapst/AdminLTEBundle/blob/master/DependencyInjection/AdminLTEExtension.php#L114) `admin_lte_theme.options` which is then [injected into the ContextHelper](https://github.com/kevinpapst/AdminLTEBundle/blob/master/Resources/config/services.yml#L20).

Using the ContextHelper instead of accessing the container parameters directly has two major advantages:
- much easier to test
- decoupling from the Symfony container

Regarding the Configuration: you have to tell Symfony, that these new config options exist, including their type and default values. Its not enough to put them into the yaml file.

Below [this line](https://github.com/kevinpapst/AdminLTEBundle/blob/master/DependencyInjection/Configuration.php#L282) you need to add something like this for each new option:
```
                ->integerNode('max_navbar_notifications')
                    ->defaultValue(5)
                    ->info('amount of notifications in the navbar dropdown')
                ->end()
...
```

Reply:
Oh and beware, I haven't tested any of the code I comment here, it might need minor adjustments... its hard to write it without IDE and auto completion^^

Reply:
Hi Kevin, I didn't forget to code. It's just that I'm having some trouble writing the tests and and finding the time to do it :)